infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/OptionalDeserializableFieldDefinition.java

```diff
@@ -40,4 +41,25 @@ public void readField(final TBuilder target, final JsonParser parser) throws IOE
     final Optional<TField> value = Optional.ofNullable(deserializableType.deserialize(parser));
     setter.accept(target, value);
   }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
Somewhat surprised we need these equals and hashCode methods for this change.  We shouldn't need to change the JSON output at all.

Reply:
Now /swagger-docs has about of dozen of duplicates in schema. Due to this. When starting swagger-ui with it, it fails to render due to schema duplicates. And btw it's incorrect json due to this, it fails on validation on json validator